src/data/zcl_abapgit_data_deserializer.clas.abap

```diff
@@ -41,23 +67,117 @@ CLASS ZCL_ABAPGIT_DATA_DESERIALIZER IMPLEMENTATION.
+    DATA:
+      lr_data TYPE REF TO data.
+
+    FIELD-SYMBOLS:
+      <lg_old> TYPE ANY TABLE,
+      <lg_new> TYPE ANY TABLE,
+      <ls_del> TYPE any,
+      <ls_ins> TYPE any,
+      <lg_del> TYPE ANY TABLE,
+      <lg_ins> TYPE ANY TABLE.
+
+    lr_data = read_database_table(
+      iv_name  = iv_name
+      it_where = it_where ).
+
+    ASSIGN lr_data->* TO <lg_old>.
+    ASSIGN ir_data->* TO <lg_new>.
+
+    rs_result-table = iv_name.
+    rs_result-deletes = zcl_abapgit_data_utils=>build_table_itab( iv_name ).
+    rs_result-inserts = zcl_abapgit_data_utils=>build_table_itab( iv_name ).
+    ASSIGN rs_result-deletes->* TO <lg_del>.
+    ASSIGN rs_result-inserts->* TO <lg_ins>.
+
+    <lg_del> = <lg_old>.
+    <lg_ins> = <lg_new>.
+
+    " Remove identical records
+    LOOP AT <lg_del> ASSIGNING <ls_del>.
+      READ TABLE <lg_ins> ASSIGNING <ls_ins> FROM <ls_del>.
+      IF sy-subrc = 0.
+        DELETE TABLE <lg_del> FROM <ls_del>.
+        DELETE TABLE <lg_ins> FROM <ls_ins>.
+      ENDIF.
+    ENDLOOP.
 
   ENDMETHOD.
 
 
-  METHOD zif_abapgit_data_deserializer~deserialize.
+  METHOD read_database_table.
+
+    DATA lv_where LIKE LINE OF it_where.
+
+    FIELD-SYMBOLS <lg_tab> TYPE ANY TABLE.
+
+    rr_data = zcl_abapgit_data_utils=>build_table_itab( iv_name ).
+    ASSIGN rr_data->* TO <lg_tab>.
 
-* this method does not persist any changes to the database
+    LOOP AT it_where INTO lv_where.
+      SELECT * FROM (iv_name) APPENDING TABLE <lg_tab> WHERE (lv_where).
+    ENDLOOP.
+    IF lines( it_where ) = 0.
+      SELECT * FROM (iv_name) INTO TABLE <lg_tab>.
+    ENDIF.
 
-    DATA lt_configs TYPE zif_abapgit_data_config=>ty_config_tt.
-    DATA ls_config LIKE LINE OF lt_configs.
-    DATA lr_data  TYPE REF TO data.
-    DATA ls_file LIKE LINE OF it_files.
+  ENDMETHOD.
 
 
+  METHOD write_database_table.
+
+    DATA:
+      lv_where   LIKE LINE OF it_where,
+      lv_tabname TYPE tabname,
+      lv_subrc   TYPE sy-subrc.
+
+    FIELD-SYMBOLS <lg_tab> TYPE ANY TABLE.
+
+    lv_tabname = iv_name.
+
+    ASSIGN ir_data->* TO <lg_tab>.
+
+    LOOP AT it_where INTO lv_where.
+      DELETE FROM (lv_tabname) WHERE (lv_where) ##SUBRC_OK.
+    ENDLOOP.
+    IF lines( it_where ) = 0.
+      CALL FUNCTION 'DB_TRUNCATE_TABLE'
+        EXPORTING
+          tabname = lv_tabname
+        IMPORTING
+          subrc   = lv_subrc.
+      IF lv_subrc <> 0.
+        zcx_abapgit_exception=>raise( |Error truncating table { lv_tabname }| ).
+      ENDIF.
+    ENDIF.
+
+    IF lines( <lg_tab> ) > 0.
+      INSERT (lv_tabname) FROM TABLE <lg_tab>.
+      IF sy-subrc <> 0.
+        zcx_abapgit_exception=>raise( |Error inserting { lines( <lg_tab> ) } records into table { lv_tabname }| ).
+      ENDIF.
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_data_deserializer~deserialize.
+
+* as a default, this method does NOT change any database tables.
+* you have to explicitly set iv_persist = abap_true to save the data
```

Comment:
https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#split-method-instead-of-boolean-input-parameter